[32m2024-05-31 12:07:24 +0000[0m - dagster - [34mDEBUG[0m - __ASSET_JOB - 11235fa8-b75c-4d89-8bc7-bf304f19d079 - 138302 - LOGS_CAPTURED - Started capturing logs in process (pid: 138302).
[32m2024-05-31 12:07:24 +0000[0m - dagster - [34mDEBUG[0m - __ASSET_JOB - 11235fa8-b75c-4d89-8bc7-bf304f19d079 - 138302 - spark_operator.create_stock_tables[stock_AACAF] - STEP_RESTARTED - Started re-execution (attempt # 4) of step "spark_operator.create_stock_tables[stock_AACAF]".
[32m2024-05-31 12:07:24 +0000[0m - dagster - [34mDEBUG[0m - __ASSET_JOB - 11235fa8-b75c-4d89-8bc7-bf304f19d079 - spark_operator.create_stock_tables[stock_AACAF] - Loading file from: /workspace/practical-data-engineering/finnhub-batch-stock-pipeline/home/storage/11235fa8-b75c-4d89-8bc7-bf304f19d079/spark_operator.stock_tables/result/stock_AACAF using PickledObjectFilesystemIOManager...
[32m2024-05-31 12:07:24 +0000[0m - dagster - [34mDEBUG[0m - __ASSET_JOB - 11235fa8-b75c-4d89-8bc7-bf304f19d079 - 138302 - spark_operator.create_stock_tables[stock_AACAF] - LOADED_INPUT - Loaded input "input_fn" using input manager "io_manager", from output "result" of step "spark_operator.stock_tables"
[32m2024-05-31 12:07:24 +0000[0m - dagster - [34mDEBUG[0m - __ASSET_JOB - 11235fa8-b75c-4d89-8bc7-bf304f19d079 - 138302 - spark_operator.create_stock_tables[stock_AACAF] - STEP_INPUT - Got input "input_fn" of type "Any". (Type check passed).
[32m2024-05-31 12:07:24 +0000[0m - dagster - [34mERROR[0m - [31m__ASSET_JOB - 11235fa8-b75c-4d89-8bc7-bf304f19d079 - 138302 - spark_operator.create_stock_tables[stock_AACAF] - STEP_FAILURE - Execution of step "spark_operator.create_stock_tables[stock_AACAF]" failed.

Exceeded max_retries of 3:

AttributeError: 'SparkSession' object has no attribute 'spark_context'. Did you mean: 'sparkContext'?

Stack Trace:
  File "/workspace/.pyenv_mirror/user/current/lib/python3.12/site-packages/dagster/_core/execution/plan/utils.py", line 54, in op_execution_error_boundary
    yield
  File "/workspace/.pyenv_mirror/user/current/lib/python3.12/site-packages/dagster/_utils/__init__.py", line 468, in iterate_with_context
    next_output = next(iterator)
                  ^^^^^^^^^^^^^^
  File "/workspace/.pyenv_mirror/user/current/lib/python3.12/site-packages/dagster/_core/execution/plan/compute_generator.py", line 141, in _coerce_op_compute_fn_to_iterator
    result = invoke_compute_fn(
             ^^^^^^^^^^^^^^^^^^
  File "/workspace/.pyenv_mirror/user/current/lib/python3.12/site-packages/dagster/_core/execution/plan/compute_generator.py", line 129, in invoke_compute_fn
    return fn(context, **args_to_pass) if context_arg_provided else fn(**args_to_pass)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workspace/practical-data-engineering/finnhub-batch-stock-pipeline/finnhub_batch_stock_pipeline/assets/spark_transformations.py", line 114, in create_stock_tables
    sc = pyspark.spark_session.spark_context
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[0m
